go-controller/pkg/ovn/pods.go

```diff
@@ -374,30 +376,44 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 			return fmt.Errorf("unable to create LSPSetDynamicAddresses command for port: %s", portName)
 		}
 		cmds = append(cmds, cmd)
-	} else {
+
+		// ensure we have/can allocate the IP in the annotation
+		if err = oc.lsManager.AllocateIPs(logicalSwitch, podIfAddrs); err != nil && err != ipallocator.ErrAllocated {
+			// this should also be treated as an allocation for purposes of error handling
+			releaseIPs = true
+			klog.Warningf("Unable to allocate IPs for already annotated pod: %s, IPs: %s, error: %v", pod.Name,
+				util.JoinIPNetIPs(podIfAddrs, " "), err)
+		} else {
+			needsIP = false
+		}
+	}
+
+	if needsIP {
+		// try to get the IP from existing port in OVN first
 		podMac, podIfAddrs, err = oc.getPortAddresses(logicalSwitch, portName)
 		if err != nil {
 			return fmt.Errorf("failed to get pod addresses for pod %s on node: %s, err: %v",
 				portName, logicalSwitch, err)
 		}
-		if podMac == nil || podIfAddrs == nil {
+		needsNewAssignment := false
+		// ensure we have/can allocate the IP found in OVN
+		if len(podIfAddrs) == 0 {
+			needsNewAssignment = true
```

Comment:
assignment -> allocation ? Perhaps we should be consistent...

Reply:
sure